### DISCUSSION TITLE
SEQURE S99

### DISCUSSION THREAD

**Discussion by jonasius** [Ralim/IronOS#1919] [General]
I just got a Sequre S99, it seems that the hardware is quite similar to the S60P. 

Wanted to backup the bootloader as statet here https://github.com/Ralim/IronOS-dfu/blob/mainline/docs/BackUp.md
I tired to use the runtime.hex from the S60P, but after transferring the file to the USB-Drive in update mode, the display switches from „UPDATE V12.00 [READY]“ to „UPDATE V12.00 [  NO  ]“

I tried to backup and afterwards flash IronOS-dfu via ST-LINK. Can’t programm the uC, also tried while in reset, it seems to be locked?!
I think I have to enter the real bootloader with BOOT0 / 1 Pins and reprogram. I’ve tried to use the commands provided [here](https://github.com/Ralim/IronOS-dfu/issues/24#issuecomment-2038135831), but only with holding NRST, which not worked.

Attached some pictures of S99 PCB+Components. 
![C304F85B-DD51-46D1-9993-23B630ABD7FB_1_201_a](https://github.com/Ralim/IronOS/assets/5968090/ba6ae478-69c4-4911-9b43-534ad23d815c)

![BFAA664F-AF36-44AD-9D24-290741785318_1_201_a](https://github.com/Ralim/IronOS/assets/5968090/153c62d0-c0cf-4656-8f3f-f79c288cb913)

I noticed that the marked resistor differs in value between S99 and S60P. If I measured correct thats the feedback resistor at the GS8332 opamp which output is connected to PA0. So it differs because the C245 tips are different and have different internal resistence for temperature measurement?!
![819365C1-CD58-44AD-820C-3F92FDD3ABD5_1_201_a](https://github.com/Ralim/IronOS/assets/5968090/7c980646-8201-490f-a591-42f3196b74ab)
![Bildschirmfoto 2024-05-25 um 12 39 22](https://github.com/Ralim/IronOS/assets/5968090/ff46a490-00f0-4261-8261-bc9b6c327be1)

**Comment by Ralim** [comment]
Hia, in future for working with the bootloader I recommend opening an issue on that repository to keep things contained.

That "NOT" means that it was flashed for the wrong starting address for the image. This changes in all of the Sequre devices seemingly at random (havent spotted the pattern yet anyway).

See if you can get a firmware update .hex file for it if you can. As that will tell you the starting location it wants (and let you flash back if you need to use the device before support is finished).

I do not recommend using a SWD programmer (such as ST-Link) until after we have a backup of the bootloader. Attempting to unlock the flash will issue a mass erase and erase all code on the device. (We need to know the bootloader address before we can add support as we want to make sure its compatible to non-modified units)


--------------


Yep that feedback resistor being different makes total sense if the tip response is different.
Likely given how similar it is; can add support fairly easily by cloning the S60P (Makefile entries/BSP folder) and then just updating the most likely different parts:

* Flash start address
* Tip uV to temp lookup (if different)
* op-amp feedback resistor value / op-amp gain

**Reply by jonasius** [reply]
Hey, yeah that makes sense.
I will open an issue regarding the bootloader in the IronOS-dfu repo. I have an update file for the firmware.

**Comment by DurvalMenezes** [comment]
@jonasius, thanks for your comments above and also for your continuing work on #1920.

I'm planning on buying my first digital soldering iron, and I'm torn between the Sequre S99 and the Pine64 Pinecil 2. From comparing specs, I see the S99 is better in terms of performance (150W vs 88W) and bang-for-the-buck ([$46 for a kit w/ 4 tips](https://www.amazon.com/SEQURE-S99-Calibration-50%C2%B0C-500%C2%B0C-Temperature/dp/B0CXPQHMVW), vs [$40 for just 1 tip](https://www.amazon.com/PINECIL-Smart-Mini-Portable-Soldering/dp/B096X6SG13)). But despite your valiant efforts (again, many thanks for that!), it seems that opensource support for the S99 is still a work in progress. 

What are your impressions of the S99 vs the Pinecil? How far do you think you are in terms of reaching operational status for the S99 on IronOS? 

TIA for your comments.

**Reply by jonasius** [reply]
Hi @DurvalMenezes 
I have no experience with the Pinecil, so I couldn't give you an advise for that. 

The PR is working for the S99, only problem is with the first version of the S60P. But that is nothing we couldn't figure out. So from that side I would say it would be save to buy a S99.  

I would assume that both irons should suit your needs if you just need a soldering iron. I like the fact that the S99 takes the JBC cartridges. So I could use the ones I already got with my JBC station. So far I didn't used the one which came with the S99.

**Reply by DurvalMenezes** [reply]
Thanks for the prompt and detailed reply, @jonasius, much obliged! 

I've decided on the S99, and I'm already procuring a way to have one delivered to me since I'm abroad ATM and have some delicate soldering work ahead of me.

Looking forward to flashing it with your PR as soon as I get it! 👍

**Reply by DurvalMenezes** [reply]
@jonasius: 
> So from that side I would say it would be save to buy a S99.

Finally my S99 reached me (took almost a month, the pleasures of being overseas). Its hardware is v1.5 (as shown in the "About" menu option: "H: 1.5").

Already tested it with the factory firmware ("R: 1.08") and it seems to be working perfectly.

So, how can I reflash it with IronOS? Where should I download the binary from? Is there installation instructions or a video somewhere?

TIA!

**Comment by kakemanx** [comment]
So on Jonasius's fork S99 is mentioned on the main page, but not (afaik) on ralims main page.. 
I cant seem to find the firmware file linked anywhere.. Is the S99 to use the same FW as the S60?

Should i even consider updating the S99 yet?... (i am a if-it-can-be-hacked-then-hack-it kinda person :/ :P )

thanks for any response to this :)

**Reply by jonasius** [reply]
The pull request is still work in progress because on older hw rev 60P irons the detection of usb-pd has some problems.
If you are feeling adeventurous, you can find the fw with support for the S99 under the action tab of the repo. But consider it in development state.

**Comment by ceut2k4** [comment]
Hello,
Some informations I forget to write here.
I have modded mine (HW 1.5) to have a properly grounded tip here:
https://www.eevblog.com/forum/reviews/sequre-s99-topic-mod-infos

Also, my mod has fixed the wrong Temperature reading of tips switching between the 5.5 and the 2.5ohms
(I understand that the opamp always uses the 5.5Ohms resistance value to read the thermocouple, I explain why on the topic).
Also, my mod allows to read the thermocouple, and put the power to the heater independently and simultaneously, so maybe it would great if this feature could be used in IronOS :-)
